src/pages/index.js

```diff
@@ -104,50 +182,118 @@ function createCard(data) {
+    });
+  })
+  .catch(err => console.error('App init error:', err));
+
+// 📝 Form Listeners
+
+// Edit profile
 document.getElementById('edit-profile-form').addEventListener('submit', e => {
   e.preventDefault();
-  profileName.textContent = profileNameInput.value;
-  profileTitle.textContent = profileDescriptionInput.value;
-  closeModal(editProfileModal);
+  const submitButton = e.submitter;
+  const updated = {
+    name: profileNameInput.value,
+    about: profileDescriptionInput.value
+  };
+  handleEditProfileSubmit(updated, submitButton);
 });
 
+// Add new card
 document.getElementById('new-post-form').addEventListener('submit', e => {
   e.preventDefault();
+  const submitButton = e.submitter;
   const newCard = {
     name: cardCaptionInput.value,
     link: cardImageInput.value
   };
-  cardsList.prepend(createCard(newCard));
+
+  setButtonText(submitButton, true, 'Saving...', 'Create');
+  api.addCard(newCard)
+    .then(card => {
+      cardsList.prepend(createCard(card));
+      closeModal(newPostModal);
+      e.target.reset();
+    })
+    .catch(err => console.error('Add card error:', err))
+    .finally(() => {
+      setButtonText(submitButton, false, 'Saving...', 'Create');
+    });
+});
+
+// Edit avatar
+editAvatarForm.addEventListener('submit', e => {
+  e.preventDefault();
+  const submitButton = e.submitter;
+  const newAvatarUrl = avatarInput.value;
+  handleEditAvatarSubmit(newAvatarUrl, submitButton);
   e.target.reset();
-  closeModal(newPostModal);
 });
 
-// Open modal event listeners
+// Modal openers
 editProfileButton.addEventListener('click', () => {
   profileNameInput.value = profileName.textContent;
   profileDescriptionInput.value = profileTitle.textContent;
   openModal(editProfileModal);
 });
 
 newPostButton.addEventListener('click', () => openModal(newPostModal));
-newPostCloseButton.addEventListener('click', () => closeModal(newPostModal));
-previewCloseButton.addEventListener('click', () => closeModal(imagePreviewModal));
-
-document.getElementById('edit-profile-form').addEventListener('submit', handleEditProfileSubmit);
-document.getElementById('new-post-form').addEventListener('submit', handleNewPostSubmit);
 
-// Close modal when clicking outside
-document.addEventListener('click', (evt) => {
-  if (evt.target.classList.contains('modal_is-opened')) {
-    closeModal(evt.target);
-  }
+avatarEditBtn.addEventListener('click', () => {
+  avatarInput.value = '';
+  openModal(editAvatarModal);
 });
 
-// Initialize
-renderInitialCards();
+// Fallback/test cards
+const initialcards = [
+  { name: 'Val Thorens', link: photo1, _id: '1' },
+  { name: 'Restaurant Terrace', link: photo2, _id: '2' },
+  { name: 'Outdoor Cafe', link: photo3, _id: '3' },
+  { name: 'Bridge in Forest', link: photo4, _id: '4' },
+  { name: 'Tunnel Morning Light', link: photo5, _id: '5' },
+  { name: 'Mountain House', link: photo6, _id: '6' }
+];
+
+initialcards.forEach(card => {
+  cardsList.appendChild(createCard(card));
```

Comment:
Please, delete `initialcards` from the project because you should get the data from the server